src/workerd/jsg/jsg-test.h

```diff
@@ -104,10 +104,15 @@ class Evaluator {
         v8::Local<v8::Value> result;
         if (script->Run(js.v8Context()).ToLocal(&result)) {
           v8::String::Utf8Value type(js.v8Isolate, result->TypeOf(js.v8Isolate));
-          v8::String::Utf8Value value(js.v8Isolate, result);
-
           KJ_EXPECT(*type == expectedType, *type, expectedType);
-          KJ_EXPECT(*value == expectedValue, *value, expectedValue);
+
+          // We ignore checking for the return value if the expected type is "undefined"
+          // because without this check, there is no way to validate a function that
+          // has a return type of undefined.
+          if (*type != "undefined"_kj && expectedValue.size() != 0) {
```

Comment:
Won't this break checks that are intentionally checking for an empty string?

I don't understand why this is needed. If you want to check for undefined, can't you do `expectEval(code, "undefined", "undefined")`?

Reply:
> for empty string you should do: expectEval(code, "string", "")

But if I did that, then this `if` statement would skip the value check. Perhaps the `&&` should have been `||`?